src/jit/function-builder.cc

```diff
@@ -362,6 +382,22 @@ bool FunctionBuilder::Emit(TR::BytecodeBuilder* b,
       Push(b, "f64", b->ConstDouble(ReadUx<double>(&pc)));
       break;
 
+    case Opcode::GetLocal:
+      // note: to work around JitBuilder's lack of support unions as value types,
+      // just copy a field that's the size of the entire union
+      Push(b, "i64", b->LoadIndirect("Value", "i64", Pick(b, ReadU32(&pc))));
+      break;
+
+    case Opcode::SetLocal:
+      // see note for GetLocal
+      b->StoreIndirect("Value", "i64", Pick(b, ReadU32(&pc)), Pop(b, "i64"));
```

Comment:
Here, you're relying on the `Pop` to happen before the `Pick` to get the right offset into the stack. However, the order of evaluation for function arguments is unspecified, so this is undefined behaviour and can break mysteriously if the compiler decides to evaluate them in the opposite order. Currently, clang does evaluate these two calls in the opposite order, emitting incorrect IL and writing to the wrong offset in the stack.